src/drive/web/modules/drive/Toolbar/components/ScanItem.jsx

```diff
@@ -0,0 +1,109 @@
+import React, { Component } from 'react'
+import { Icon, translate } from 'cozy-ui/transpiled/react'
+
+import Scanner from './Scanner'
+import toolbarContainer from '../toolbar'
+import PortaledQueue from './PortaledQueue'
+
+import {
+  startMediaBackup,
+  cancelMediaBackup
+} from 'drive/mobile/modules/mediaBackup/duck'
+
+import { connect } from 'react-redux'
+/**
+ *
+ */
+
+const ScanItemMenu = translate()(({ status, onClick, t, online }) => {
+  const offlineMessage = () => {
+    return alert(t('Scan.error.offline'))
+  }
+  const uploadingMessage = () => {
+    return alert(t('Scan.error.uploading'))
+  }
+  const actionOnClick = (() => {
+    if (status === 'uploading') return uploadingMessage
+    if (!online) return offlineMessage
+    return onClick
+  })()
+
+  return (
+    <span className="u-pl-1 u-flex u-pt-half u-pb-half" onClick={actionOnClick}>
```

Comment:
Shouldn't this be `actionOnClick()`?

Reply:
Nop. We invoke immediately the method instancing `actionOnClick`. So in `actionOnClick` it's bind to the right method. No need to invoke it again. `onClick` will trigger the invokation